src/ToastProvider.js

```diff
@@ -72,19 +76,30 @@ type Context = {
 
 export class ToastProvider extends Component<Props, State> {
   static defaultProps = {
+    name: DEFAULT_CONTEXT_NAME,
     autoDismiss: false,
     autoDismissTimeout: 5000,
     components: defaultComponents,
     placement: 'top-right',
     transitionDuration: 220,
   };
 
-  state = { toasts: [] };
+  constructor(props: Props) {
+    super(props);
+
+    contexts[props.name] = contexts[props.name] || React.createContext();
+    const context = contexts[props.name];
```

Comment:
We only create a new context if there isn't already one for that name.

Then we continue using the created context within this component for its lifetime.